### PR TITLE
javascript/linked-list: minimal mentor notes

### PR DESCRIPTION
**Pull request by SleeplessByte**
Add minimal mentor notes for `linked-list`

### CODE REVIEW

File: tracks/javascript/exercises/linked-list/mentoring.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+      break
+    }
+  }
+}
+```
+
+Variations include the following
+- A student may only pass `value` to `Node` and assign `prev` and `next` afterwards
+- A student may forego the `throw` clauses completely as there are no test cases covering empty list popping and shifting.
+- A student may actually put the logic on the `Node` class (classic linked list) and only have the `LinkedList` API keep track of `head` and `tail` (and fix them on mutations). `LinkedList` is than a mere proxy to the actual doubly linked-list.
```
Comment (tejasbubane) [inline tracks/javascript/exercises/linked-list/mentoring.md]: Do you mean to say `is there as a mere proxy ...` ?

File: tracks/javascript/exercises/linked-list/mentoring.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+Variations on this include array destructuring in the implementation of `pop`,
+`push` etc, or using a backing array inside the `LinkedList` class. See *common
+suggestions* for more information.
+
+#### Note about earlier versions
+Earlier version had a different `Readme.MD` that pointed to using a `Deque`.
+This usually only confused students and has been removed.
+
+### Common suggestions
+- If a student uses a backing array or the likes, talk about performance and the upsides of actually using a linked-list.
```
Comment (tejasbubane) [inline tracks/javascript/exercises/linked-list/mentoring.md]: Also talk about `linked lists` being [classical data structures](https://en.wikipedia.org/wiki/Linked_data_structure) and it is a good exercise to implement them in the intended way.

**Comment by SleeplessByte** [comment]
Woops. Yes. Thanks <3

**Comment by iHiD** [comment]
Let's get this merged down then move forward on master?

**Pull request merged**